keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -649,89 +629,182 @@ endpoints:
+  Each `EndpointSlice` associated with the exported service is combined with
+  slices from other clusters to make up the supercluster service. They will be
+  imported to the cluster behind the supercluster IP, with a `ServiceImport` of
+  type `SuperclusterIP`. [Details](#EndpointSlice)
+- `ClusterIP: none` (Headless): Headless services are supported and will be
+  imported with a `ServiceImport` and `EndpointSlices` like any other
+  `ClusterIP` service, but do not configure a VIP and must be consumed via
+  [DNS](#DNS). Their `ServiceImport`s will be of type `Headless`. A
+  multi-cluster service's headlessness is derived from it's constituent exported
+  services according to the [conflict resolution
+  policy](#constraints-and-conflict-resolution).
+
+  _Exporting a non-headless service to an otherwise headless service can
+  dynamically change the supercluster service type when an old export is
+  removed, potentially breaking existing consumers. This is likely the result of
+  a deployment error. Conditions and events on the `ServiceExport` will be used
+  to communicate conflicts to the user._
+- `NodePort` and `LoadBalancer`: These create `ClusterIP` services that would
+  sync as expected. For example If you export a `NodePort` service, the
+  resulting cross-cluster service will still be a supercluster IP type. The
+  local service will not be affected. Node ports can still be used to access the
+  cluster-local service in the source cluster, and only the supercluster IP will
+  route to endpoints in remote clusters.
+- `ExternalName`: It doesn't make sense to export an `ExternalName` service.
+  They can't be merged with other exports, and it seems like it would only
+  complicate deployments by even attempting to stretch them across clusters.
+  Instead, regular `ExternalName` type `Services` should be created in each
+  cluster individually. If a `ServiceExport` is created for an `ExternalName`
+  service, an `InvalidService` condition will be set on the export.
+
+#### SuperclusterIP
+
+A non-headless `ServiceImport` is expected to have an associated IP address, the
+supercluster IP, which may be accessed from within an importing cluster. This IP
+may be a single IP used supercluster-wide or assigned on a per-cluster basis,
+but is expected to be consistent for the life of a `ServiceImport` from the
+perspective of the importing cluster. Requests to this IP from within a cluster
+will route to backends for the aggregated Service.
+
+Note: this doc does not discuss `NetworkPolicy`, which cannot currently be used
+to describe a selector based policy that applies to a multi-cluster service.
+
+#### DNS
+
+_Optional, but recommended._
+
+MCS aims to align with the existing [service DNS
+spec](https://github.com/kubernetes/dns/blob/master/docs/specification.md). This
+section assumes familiarity with in-cluster Service DNS behavior.
+```
+<<[UNRESOLVED]>>
+Full DNS spec needed prior to Beta graduation following:
```

Comment:
We should plan for a deep design discussion ASAP but I won't block it now, as agreed previously. :)

Reply:
Yeah, as soon as this is accepted I'll start setting up a deep dive. There's a lot to consider here and it will likely drive some of what we need out of Cluster ID